pandas/core/generic.py

```diff
@@ -2420,6 +2451,12 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
             (default is False)
         compress : type of compressor (zlib or blosc), default to None (no
             compression)
+
+        Returns
+        -------
+        None or str
```

Comment:
Just a preference, but I'd prefer `str or None`

Reply:
It is written in the same way as is in other to_* io functions. It returns str only when a path or buffer-like is not provided. I believe it mostly returns None which makes writing `None or str` more reasonable IMO.